libtenzir/builtins/operators/every_cron.cpp

```diff
@@ -218,6 +233,49 @@ class every_plugin final : public virtual operator_plugin<every_operator> {
   }
 
   auto parse_operator(parser_interface& p) const -> operator_ptr override {
+    using operator_type = scheduled_execution_operator<Scheduler>;
+    auto scheduler = Scheduler::parse(p);
+
+    auto result = p.parse_operator();
+    if (not result.inner) {
+      diagnostic::error("failed to parse operator")
+        .primary(result.source)
+        .throw_();
+    }
+    if (auto* pipe = dynamic_cast<pipeline*>(result.inner.get())) {
+      auto ops = std::move(*pipe).unwrap();
+      for (auto& op : ops) {
+        op = std::make_unique<operator_type>(std::move(op),
+                                             std::move(scheduler));
+        // Only the first operator can be a source and needs to be replaced.
+        break;
+      }
+      return std::make_unique<pipeline>(std::move(ops));
+    }
+    return std::make_unique<operator_type>(std::move(result.inner),
+                                           std::move(scheduler));
+  }
+};
+
+/// scheduler for the 'every' operator
+class every_scheduler {
+public:
+  every_scheduler() = default;
+  explicit every_scheduler(const duration interval) : interval_{interval} {
+  }
```

Comment:
With what rule do you decide between `const` and non-`const` value parameters? This looks pretty arbitrary to me. (I personally never use `const` in this situation, but my point is: consistency is king).

Reply:
Basically I mark all parameters as `const` if I dont intend to modify/move from them - unless I forget of course. 

I can already see that i forgot those in my attempts to make `next_after` work on mac.